src/components/Dashboard.tsx

```diff
@@ -1,46 +1,90 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react'
+import logo from '../../public/images/logo.PNG'
 import Image from 'next/image';
-import { UserData, ComponentProps } from '@/types/types';
-import { getTodaysDate } from '@/utils';
+import { getTodaysDate, getZodiacSign } from '@/utils';
 import Navbar from '../components/Navbar';
 import { insights } from '@/mockdata';
 import Router from "next/router";
-import Logo from '@/components/logo';
+import { Passage } from '@passageidentity/passage-js';
+import { PassageUserInfo } from '@passageidentity/passage-elements/passage-user';
+import dotenv from 'dotenv';
+import { AuthProps } from '@/types/types';
 
+const getCurrentUser = async () => {
+  dotenv.config();
+  const passageAppId = process.env.NEXT_PUBLIC_PASSAGE_APP_ID;
 
-type DashboardProps = ComponentProps & {
-  logOut: () => void
-};
+  if (!passageAppId) {
+    throw new Error('NEXT_PUBLIC_PASSAGE_APP_ID is not defined in the environment.');
+  }
 
+  const passage = new Passage(passageAppId);
+  const user = passage.getCurrentUser();
+  const userInfo = await user.userInfo();
+  return userInfo
+}
```

Comment:
The userID is already provided by the `getServerSideProps` in this component

Reply:
ahh good point! I got those mixed up a lot looking at the different docs

Reply:
The docs can be super confusing 🫨 especially to us BE newbs

Reply:
After looking over this again, this is to retrieve the user data from Passage, so it would be different from supabase since supabase doesn't have the user data at this point in the code

Reply:
*the userID is the same but this function is returning all of the user's data: 
```
interface PassageUserInfo {
    created_at: string;
    update_at: string;
    status: string;
    id: string;
    email: string;
    email_verified: boolean;
    phone: string;
    phone_verified: boolean;
    webauthn: boolean;
    webauthn_devices: PassageDevice[];
    last_login_at: string;
    login_count: number;
    user_metadata: Metadata | null;
    webauthn_types: WebAuthnType[];
}